### PR TITLE
Fix exception thrown when nonexistent filename, especially `-` is passed

### LINKED ISSUE #44: flake8helled vs stdin
``flake8 --stdin-display-name filename - < file.py`` does allow to feed flake8 via stdin while seeing a given filename (rather than "stdin") in the report
``flake8helled --stdin-display-name filename - < file.py`` doesn't work cause it interprets the "-" as a file in the current directory, given the following error:
``FileNotFoundError: [Errno 2] No such file or directory: '/path/to/current/directory/-'``

I didn't investigate further for now, and I'm afraid that it's beyond my competence. Or is there an obvious solution?
Thanks for attention :)

### PR DESCRIPTION
**Pull request by k3rni**
The error is caused by Snapshot class naively reading in file content to
create a md5 digest for the file. This is not possible when `-` is passed,
because a) it is most likely not an existing filename, and b) even if we
were to interpret it as stdin, it isn't seekable (outside of using `os.lseek` when a real file was redirected).

The fix: when calculating digests for a non-existent file, just invent
one from current time and a random value. It will be a new value every
time, ensuring no caching between such stdin invocations.

Fixes #44

### CODE REVIEW

**Comment by k3rni** [comment]
The build errors on this are _fun_. Is the build process broken?

**Comment by k3rni** [comment]
Need help understanding the build failures :confused:

**Comment by bid0uille** [comment]
Thanks for the insight. I will give a try to your fix.

**Comment by orsinium** [comment]
Thank you!

File: tests/test_flake8helled_stdin.py [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
 
 
 def test_flake8helled_file():
     """Baseline behavior, when an actual filename is passed."""
-    cmd = ['python',
-           # This is what flit creates as the 'flake8helled' command
-           '-c', 'import sys; from flakehell import flake8_entrypoint; sys.exit(flake8_entrypoint())',
-           __file__]
+    cmd = [
+        sys.executable,
```
Comment (k3rni) [inline tests/test_flake8helled_stdin.py:8]: :+1:

**Pull request merged**